src/include/txn_inline.h

```diff
@@ -917,8 +917,12 @@ __wt_upd_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value, u_int modify_type
      *    WT_UPDATE.start_ts = WT_TS_NONE;
      *    WT_UPDATE.prepare_state = WT_PREPARE_INIT;
      *    WT_UPDATE.flags = 0;
+     *
+     * When there is no value data, use sizeof() which will include any (trailing) padding bytes
+     * added by the compiler.
      */
-    WT_RET(__wt_calloc(session, 1, WT_UPDATE_SIZE + (value == NULL ? 0 : value->size), &upd));
+    WT_RET(__wt_calloc(
+      session, 1, (value == NULL) ? sizeof(*upd) : (WT_UPDATE_SIZE + value->size), &upd));
```

Comment:
After reading through the comments of the ticket, the memory for the last member of the `upd->data` is not allocated leading to the problem. Using sizeof(*upd) currently yields 48 bytes when there is no value, in future if the update structure is modified, it can lead to a different value. Instead of `sizeof` how about using `WT_UPDATE_SIZE + 1`?

Reply:
This is a speculative fix. The idea is to ensure that any trailing space in the structure will be allocated (though not used) and initialized. If `WT_UPDATE_SIZE + 1` were used it would break / need to be changed depending on any how changing WT_UPDATE impacts the amount of trailing space in the structure.
That said thinking about that scenario this change would not handle that situation correctly if `WT_UPDATE_SIZE + value->size < sizeof(WT_UPDATE_SIZE)`. So I will change.